### PR TITLE
Adding service mocks and service unit tests

### PR DESCRIPTION
**Pull request by uristemov**
Adding service mocks and service unit tests;
Deleting unused comments;
Correcting codes;

### CODE REVIEW

File: api/author.go [hunk 1]
```diff
@@ -3,9 +3,9 @@ package api
 import "github.com/google/uuid"
 
 type AuthorRequest struct {
-	Id          *uuid.UUID `json:"author_id"`
-	Firstname   *string    `json:"firstname"`
-	Lastname    *string    `json:"lastname"`
-	ImagePath   *string    `json:"image_path"`
-	AboutAuthor *string    `json:"about_author"`
+	Id          uuid.UUID `json:"author_id"`
```
Comment (murmurat) [inline api/author.go]: Id is never used, we can delete this field

File: internal/handler/book.go [hunk 2]
```diff
@@ -3,104 +3,108 @@ package handler
...
 
 	err := h.srvs.CreateBook(ctx, &req)
 	if err != nil {
-		log.Printf("Error %w", err)
-		ctx.JSON(http.StatusInternalServerError, err)
+		ctx.AbortWithStatusJSON(http.StatusInternalServerError, api.Error{Message: err.Error()})
+		return
 	}
 
 	ctx.JSON(http.StatusCreated, req)
```
Comment (murmurat) [inline internal/handler/book.go]: I think there's no need to return the request in response

Comment (murmurat) [reply internal/handler/book.go]: Returning ID makes more sense (just for future). But for now we can return just empty 201 response

File: internal/handler/router.go [hunk 3]
```diff
@@ -15,29 +15,40 @@ func (h *Handler) InitRouter() *gin.Engine {
...
-	router.PUT("/file_path/update/:id", h.updateFilePath)    //checked
-
-	router.PUT("/user/update", h.updateUser)
-	router.DELETE("/user/delete", h.deleteUser)
-	router.GET("/user/data", h.getUser)
+	user := router.Group("/user", h.authMiddleware())
+	{
+		user.PUT("/update", h.updateUser)
+		user.DELETE("/delete", h.deleteUser)
+		user.GET("/data", h.getUser)
```
Comment (murmurat) [inline internal/handler/router.go:22]: "/user/books" should be here i think

**Review by murmurat — CHANGES_REQUESTED** [review]

**Comment by uristemov** [comment]
Fixed errors in service mocks and unit tests
Working on murat96k comments
correcting createUser() errors
regenerate mocks for new createEntity codes
adding sql transactions on create method (Begin, Rollback, Commit)

**Review by murmurat — APPROVED** [review]

**Pull request merged**